service-pdf/tests/OpgTest/Lpa/Pdf/ContinuationSheet2Test.php

```diff
@@ -0,0 +1,640 @@
+    }
+
+    public function testGenerateMultiRAsWhenLastHowDepends()
+    {
+        $lpa = $this->getLpa();
+
+        //  Set up the content
+        //  Adapt the LPA as required
+        $lpa->document->replacementAttorneyDecisions = [
+            'when'        => ReplacementAttorneyDecisions::LPA_DECISION_WHEN_LAST,
+            'how'         => ReplacementAttorneyDecisions::LPA_DECISION_HOW_DEPENDS,
+            'howDetails'  => 'test test',
+        ];
+
+        $content = $this->getHowWhenReplacementAttorneysCanActContent($lpa->document);
+
+        $pdf = new ContinuationSheet2($lpa, ContinuationSheet2::CS2_TYPE_REPLACEMENT_ATTORNEYS_STEP_IN, $content, 1);
+
+        //  Set up the expected data for verification
+        $formattedLpaRef = 'A510 7295 5715';
+        $templateFileName = 'LPC_Continuation_Sheet_2.pdf';
+
+        $strikeThroughTargets = [];
+
+        $blankTargets = [];
+
+        $constituentPdfs = [
+            'start' => [
+                $this->getFullTemplatePath('blank.pdf'),
+            ],
+        ];
+
+        $data = [
+            'cs2-is' => "how-replacement-attorneys-step-in",
+            'cs2-content' => "\r\nReplacement attorneys to step in only when none of the original attorneys can act   \r\nReplacement attorneys are to act joint for some decisions, joint and several for    \r\nother decisions, as below:                                                          \r\ntest test                                                                           ",
+            'cs2-donor-full-name' => "Mrs Nancy Garrison",
+            'cs2-continued' => "",
+            'cs2-footer-right' => "LPC Continuation sheet 2 (07.15)",
+        ];
+
+        $pageShift = 0;
+
+        $this->verifyExpectedPdfData($pdf, $templateFileName, $strikeThroughTargets, $blankTargets, $constituentPdfs, $data, $pageShift, $formattedLpaRef);
+
+        //  Test the generated filename created
+        $pdfFile = $pdf->generate();
+
+        $this->verifyTmpFileName($lpa, $pdfFile, 'ContinuationSheet2.pdf');
+    }
+
+    public function testGenerateMultiRAsWhenLastHowJointlyAndSeverally()
+    {
+        $lpa = $this->getLpa();
+
+        //  Set up the content
+        //  Adapt the LPA as required
+        $lpa->document->replacementAttorneyDecisions = [
+            'when'        => ReplacementAttorneyDecisions::LPA_DECISION_WHEN_LAST,
+            'how'         => ReplacementAttorneyDecisions::LPA_DECISION_HOW_JOINTLY_AND_SEVERALLY,
+            'howDetails'  => 'test test',
+        ];
+
+        $content = $this->getHowWhenReplacementAttorneysCanActContent($lpa->document);
+
+        $pdf = new ContinuationSheet2($lpa, ContinuationSheet2::CS2_TYPE_REPLACEMENT_ATTORNEYS_STEP_IN, $content, 1);
+
+        //  Set up the expected data for verification
+        $formattedLpaRef = 'A510 7295 5715';
+        $templateFileName = 'LPC_Continuation_Sheet_2.pdf';
```

Comment:
Same again with this value. In fact, for all the good path tests values in the pdf moving these into `AbstractPdfTestClass` & then only over-ridding them when needed for bad path or testing the limits? Just so we can cut down a lot on the bloat in these files & make it a little easier to parse.

Reply:
I've gone through and removed as much repetition as I can.